engine/apps/public_api/serializers/schedules_base.py

```diff
@@ -1,83 +1,66 @@
 
-    def get_on_call_now(self, obj):
+    def get_on_call_now(self, obj: "OnCallSchedule") -> typing.List[str]:
         users_on_call = list_users_to_notify_from_ical(obj, datetime.datetime.now(datetime.timezone.utc))
         if users_on_call is not None:
             return [user.public_primary_key for user in users_on_call]
         else:
             return []
 
     def _correct_validated_data(self, validated_data):
-        slack_field = validated_data.pop("slack", {})
-        if "channel_id" in slack_field:
-            validated_data["channel"] = slack_field["channel_id"]
+        if slack_field := validated_data.pop("slack", {}):
+            if "channel_id" in slack_field:
+                validated_data["slack_channel"] = slack_field["channel_id"]
 
-        if "user_group_id" in slack_field:
-            validated_data["user_group"] = SlackUserGroup.objects.filter(slack_id=slack_field["user_group_id"]).first()
+            if "user_group_id" in slack_field:
+                validated_data["user_group"] = slack_field["user_group_id"]
 
         return validated_data
 
-    def validate_slack(self, slack_field):
-        from apps.slack.models import SlackChannel
-
-        slack_channel_id = slack_field.get("channel_id")
-        user_group_id = slack_field.get("user_group_id")
-
-        organization = self.context["request"].auth.organization
-        slack_team_identity = organization.slack_team_identity
-
-        if (slack_channel_id or user_group_id) and not slack_team_identity:
-            raise BadRequest(detail="Slack isn't connected to this workspace")
-
-        if slack_channel_id is not None:
-            slack_channel_id = slack_channel_id.upper()
-            try:
-                slack_team_identity.get_cached_channels().get(slack_id=slack_channel_id)
-            except SlackChannel.DoesNotExist:
-                raise BadRequest(detail="Slack channel does not exist")
-
-        if user_group_id is not None:
-            user_group_id = user_group_id.upper()
-            try:
-                slack_team_identity.usergroups.get(slack_id=user_group_id)
-            except SlackUserGroup.DoesNotExist:
-                raise BadRequest(detail="Slack user group does not exist")
-
-        return slack_field
-
-    def to_representation(self, instance):
-        result = super().to_representation(instance)
-
-        user_group_id = instance.user_group.slack_id if instance.user_group is not None else None
-        result["slack"] = {
-            "channel_id": instance.channel or None,
-            "user_group_id": user_group_id,
+    def to_representation(self, instance: "OnCallSchedule"):
+        return {
+            **super().to_representation(instance),
+            "slack": {
+                "channel_id": instance.slack_channel_slack_id,
+                "user_group_id": instance.user_group.slack_id if instance.user_group is not None else None,
+            },
         }
 
-        return result
```

Comment:
see `SlackSerializer` above.. all of the validation logic is deduplicated (and tested) in those custom serializers